src/gui/qgsmaptooladvanceddigitizing.cpp

```diff
@@ -25,7 +25,6 @@ QgsMapToolAdvancedDigitizing::QgsMapToolAdvancedDigitizing( QgsMapCanvas *canvas
   : QgsMapToolEdit( canvas )
   , mCadDockWidget( cadDockWidget )
 {
-  Q_ASSERT( cadDockWidget );
```

Comment:
What's the reason for removing this?

Reply:
I want to disable geometry creation from relation reference widget and relation editor based on the fact that _cadDockWidget_ variable from context is null because of QgsMapToolFillRing (it creates one feature and modify another, it's a mess if you start creating child at this very moment).

After a second thoughts, I change the way I did this to keep the assert. Do you think it's fine this way?

Reply:
Yes, that sounds good. I'd just add a docstring (to `setEditorContext`) to make this kind of behavior explicit for other places where we will want the same behavior.

Reply:
Done! I also add the same kind of behavior for the relation reference widget.